marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/default/lib.sjs

```diff
@@ -128,7 +128,7 @@ function extractInstanceFromModel(model, modelName, mapping, content, provenance
     }
     let value = null;
     if (!dataType && prop['$ref']) {
-      let refArr = prop['$ref'].split('/');
+      let refArr = String(prop['$ref']).split('/');
```

Comment:
What change is causing a String cast to be needed here and in the places below?

Reply:
Not exactly a change. The OpenApi specs were wrong and had "targetEntity" instead of "targetEntityType". That was hitting the split error since targetEntityType was null. The cast was part of the debugging. I left it since the error "Couldn't find targetEntityType: " more clearly indicated what was wrong than the split undefined error we we seeing.

Reply:
@ryanjdew, some tests are failing with missing ["targetEntityType"] error.

Reply:
@bsrikan addressing the issues now